Change bits service from loadbalanced to nodePort.

andreas-kupries opened the pull request:
## Description

Removed `loadbalanced` property.
Added `nodePort` property.
Changed templates to trigger on `nodePort`. Removed all references to `loadbalanced`.

## Motivation and Context

See https://github.com/cloudfoundry-incubator/kubecf/issues/792

## How Has This Been Tested?

Local minikube, SA eirini. Smoke tests run manually.

andreas-kupries committed:
[gh#kubecf-792] Change bits service from loadbalanced to nodePort.

mook-as commented on helm/bits/templates/bits-config.yaml:
```diff
@@ -50,7 +50,7 @@ stringData:
         public_endpoint: https://blobstore.{{ .Values.ingress.endpoint }}
-        {{- else if .Values.services.loadbalanced }}
+        {{- else if .Values.services.nodePort }}
         public_endpoint: https://blobstore.{{ .Values.env.DOMAIN }}
```
That seems odd: if we're using nodeport, surely the port should be in there somewhere? (Yes that means that we will need to adapt the kubecf part to remove some of the hacks.)

andreas-kupries replied:
Hm ... Oh. Yes, we currently have the port info in kubecf, see https://github.com/cloudfoundry-incubator/kubecf/blob/56f87f06fdedf93929d43f196af36084bc48be5f/deploy/helm/kubecf/values.yaml#L373-L376

Will fix, i.e. shift the part from kubecf to here.

andreas-kupries replied:
Done with commit 1e05c56e27fc4fd4b7d4d3ac2226306029f0f315
and https://github.com/cloudfoundry-incubator/kubecf/pull/1022/commits/55ab51b135d41881af2cc8141cbd00e147e564e9

andreas-kupries committed:
Address comments.

Make nodeport explicit in the endpoint definitions.
Fix missing exclusion of nodeport when not defined.

mook-as reviewed (approved):
Sorry for the delay; LGTM.

mook-as commented:
Hmm, I'm having issues using this in conjunction with https://github.com/cloudfoundry-incubator/kubecf/pull/1022 — it looks like eirini is expecting bits to be on port 6666. Not quite sure _where_ fix should be, though.

viovanov commented:
@mook-as the fix should be in the eirini chart iirc - if it doesn't allow setting the port
This looks ok, I'll merge.
